### PR TITLE
CORTX-333872 : F-40E 2 Tests TA

### PR DESCRIPTION
**Pull request by RajeshPratapSInghChouhan**
# Problem Statement
[CORTX-33387](https://jts.seagate.com/browse/CORTX-33387) TA : F-40E CSM : Deployment: Config: Resource Limits Part-2

# Design
https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1077084998/QA+F-40E+CSM+Resource+Limits+tests

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [x] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [x] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/csm/rest/test_csm_load.py [hunk 1]
```diff
@@ -118,6 +167,160 @@ def test_poc(self):
...
+        result = self.jmx_obj.run_verify_jmx(
+            jmx_file,
+            threads=self.request_usage,
+            rampup=test_cfg["rampup"],
+            loop=test_cfg["loop"])
+        assert result, "Errors reported in the Jmeter execution"
+
+        stop_res = run_data_chk_obj.stop_io_async(users=data, di_check=True, eventual_stop=True)
+        self.log.info("stop_res -  %s", stop_res)
+
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_csm_load.py:215]: Does this function assures that IO are running properly in background?

File: tests/csm/rest/test_csm_load.py [hunk 2]
```diff
@@ -118,6 +167,160 @@ def test_poc(self):
...
+        resp = self.csm_obj.get_user_capacity_usage("user", username)
+        assert resp.status_code == HTTPStatus.OK, \
+                "Status code check failed for get capacity"
+        t_obj = resp.json()["capacity"]["s3"]["users"][0]["objects"]
+        t_size = resp.json()["capacity"]["s3"]["users"][0]["used"]
+        m_size = resp.json()["capacity"]["s3"]["users"][0]["used_rounded"]
+        self.log.info("objects -  %s", t_obj)
+        self.log.info("used capacity-  %s", t_size)
+        self.log.info("used_rounded capacity-  %s", m_size)
+        assert t_obj != 0, "Number of objects is Zero"
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_csm_load.py:226]: t_obj > 0

**Review by kachhwahadivya — APPROVED** [review]

File: tests/csm/rest/test_csm_load.py (hunk 2 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_load.py]: updated

**Comment by RajeshPratapSInghChouhan** [comment]
#pytest --local True --collect-only
r_netstat', 'test_id': 'TEST-34218', 'marks': ['security']}] created at /root/jenkins/workspace/QA/QA_R2_Single_Runner_CSM_error_msg_check/cortx-test/log/te_meta.json
Successfully unmounted directory

================================================================================= 2384 tests collected in 11.80s =================================================================================
[root@ssc-vm-g2-rhev4-1909 cortx-test]# git branch
* CORTX-333872_TA_F-40E_Tests

File: tests/csm/rest/test_csm_load.py [hunk 3]
```diff
@@ -70,21 +75,56 @@ def setup_class(cls):
         assert s3acc_already_present
         cls.log.info("[Completed]: Setup class")
         cls.default_cpu_usage = False
+        cls.buckets_created = []
         cls.iam_users_created = []
+        cls.ha_obj = HAK8s()
+        cls.failed_pod = []
+        cls.restore_pod = cls.deployment_backup = cls.deployment_name = cls.restore_method = None
+        cls.num_replica = 1
```
Comment (sonalk0209) [inline tests/csm/rest/test_csm_load.py:83]: will this always be 1? or default settings are for replicaset?

File: tests/csm/rest/test_csm_load.py [hunk 4]
```diff
@@ -118,6 +165,161 @@ def test_poc(self):
...
+        delete_pod = sts_dict[sts][-1]
+        self.log.info("Pod to be deleted is %s", delete_pod)
+        set_type, set_name = self.csm_obj.master.get_set_type_name(pod_name=delete_pod)
+        if set_type == const.STATEFULSET:
+            resp = self.csm_obj.master.get_num_replicas(set_type, set_name)
+            assert resp[0], resp
+            self.num_replica = int(resp[1])
+            num_replica = self.num_replica - 1
+
+        self.log.info("Shutdown random data pod by making replicas=0 and "
```
Comment (sonalk0209) [inline tests/csm/rest/test_csm_load.py:266]: we are not making replicas=0 with stafefulset, so u can just say with replica method

File: tests/csm/rest/test_csm_load.py (hunk 3 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_load.py]: removed

File: tests/csm/rest/test_csm_load.py (hunk 4 above)
Comment (RajeshPratapSInghChouhan) [reply tests/csm/rest/test_csm_load.py]: updated

**Review by aditipande96 — APPROVED** [review]

**Comment by aditipande96** [comment]
Please resolve all comments on PR.

**Review by sonalk0209 — APPROVED** [review]

**Pull request merged**